l2os/drivers/l2output/driver.go

```diff
@@ -196,7 +196,12 @@ func (d *Driver) CraftTx(
 
 	numElements := new(big.Int).Sub(start, end).Uint64()
 	log.Info(name+" checkpoint constructed", "start", start, "end", end,
-		"nonce", nonce, "blocks_committed", numElements)
+		"nonce", nonce, "blocks_committed", numElements, "checkpoint_block", nextCheckpointBlock)
+
+	header, err := d.cfg.L1Client.HeaderByNumber(ctx, nil)
```

Comment:
@cfromknecht: I believe that this is the correct logic, but I'm not sure. Would love a gut check.

Reply:
@mslipper I suspect we want to pass `nextCheckpointBlock` rather than `nil`, to make sure the header hash corresponds to the L2Output being submitted?

Reply:
We can't pass `nextCheckpointBlock` because the sequencer will always be super far ahead of L1. Is there a different block we should be using?

Reply:
ah i see, missed that this is an L1 block. reviewed the contract again looks good to me 👍